### PR TITLE
Addressing #379

### LINKED ISSUE #379: Files Resolve Path
#375 / #378 Added support for `GET /api/v1/courses/:course_id/folders/by_path/*full_path`, but there are several other related endpoints under the "Resolve Path" umbrella.

**What resource needs additional coverage?**
Files, Folders, "Resolve Path"

**What endpoints need to be covered?**

- [ ] `GET /api/v1/courses/:course_id/folders/by_path`
- [ ] `GET /api/v1/users/:user_id/folders/by_path/*full_path`
- [ ] `GET /api/v1/users/:user_id/folders/by_path`
- [ ] `GET /api/v1/groups/:group_id/folders/by_path/*full_path`
- [ ] `GET /api/v1/groups/:group_id/folders/by_path`

https://canvas.instructure.com/doc/api/files.html#method.folders.resolve_path

### PR DESCRIPTION
**Pull request by dsavransky**
# Proposed Changes

Expanding course.resolve_path to accept null input thereby covering the `GET /api/v1/courses/:course_id/folders/by_path` endpoint as well as the existing `GET /api/v1/courses/:course_id/folders/by_path/*full_path` coverage.

Cloning identical coverage to `user` and `group` classes.  All resolve_path endpoints should now be covered.

Updating unit tests for all 3 classes.

Fixes #379

### CODE REVIEW

File: tests/test_user.py [hunk 1]
```diff
@@ -625,6 +625,25 @@ def test_get_licenses(self, m):
 
         self.assertEqual(2, len(licenses))
 
+    # resolve_path()
```
Comment (jessemcbride) [inline tests/test_user.py:628]: It looks like there may be separate test cases in here. Can we split it into two functions (e..g `test_resolve_path_null_input` and `test_resolve_path`?

File: tests/test_user.py [hunk 2]
```diff
@@ -625,6 +625,25 @@ def test_get_licenses(self, m):
...
+    def test_resolve_path(self, m):
+        register_uris({"user": ["resolve_path_null", "resolve_path"]}, m)
+
+        full_path = "Folder_Level_1/Folder_Level_2/Folder_Level_3"
+        folders = self.user.resolve_path(full_path)
+        folder_list = [folder for folder in folders]
+        self.assertEqual(len(folder_list), 4)
+        self.assertIsInstance(folder_list[0], Folder)
+        for folder_name, folder in zip(("my_files/" + full_path).split("/"), folders):
```
Comment (jessemcbride) [inline tests/test_user.py:637]: The `zip` call makes the Iterable in this for-loop pretty tricky to read - could we pull the resulting list out into its own variable?

Comment (dsavransky) [reply tests/test_user.py:637]: Do you mean pre-defining the `folder_name` list?  So something like:
```
folder_names = "my_files/" + full_path).split("/")
for j in range(len(folder_names):
     folder_name = folder_names[j]
     folder = folders[j]
...
```

Comment (jessemcbride) [reply tests/test_user.py:637]: Yeah, pulling out `folder_names = (my_files/" + full_path).split("/") and then iterating with `for folder_name, folder in zip(folder_names, folders)` seems fine to me.

**Comment by Thetwam** [comment]
Thanks for the PR, @dsavransky 😁

**Pull request merged**